pkgs/by-name/ar/artisan/package.nix

```
@@ -1,28 +1,136 @@
 {
   lib,
-  appimageTools,
-  fetchurl,
-  nix-update-script,
+  stdenv,
+  fetchFromGitHub,
+
+  python3,
+  makeWrapper,
+  wrapGAppsHook,
+  qt6,
+  qt6Packages,
+
+  libusb1,
+  libphidget22,
+  libphidget22extra,
+
 }:
 let
-  pname = "artisan";
+
   version = "3.1.4";
+  pname = "artisan";
 
-  src = fetchurl {
-    url = "https://github.com/artisan-roaster-scope/artisan/releases/download/v${version}/${pname}-linux-${version}.AppImage";
-    hash = "sha256-GU9DCAIMioPRUC1IDu8SbhD802pWEH7MlALnQH+Y4oU=";
-  };
+  inherit (qt6Packages) wrapQtAppsHook;
 
-  appimageContents = appimageTools.extract {
-    inherit pname version src;
-  };
+  pythonEnv = python3.withPackages (
+    ps: with ps; [
+      phidget22
+      yoctopuce
+      setuptools
+      wheel
+      pyserial
+      pymodbus
+      python-snap7
+      unidecode
+      qrcode
+      requests
+      requests-file
+      pyusb
+      portalocker
+      xlrd
+      websockets
+      pyyaml
+      psutil
+      typing-extensions
+      protobuf
+      numpy
+      scipy
+      openpyxl
+      keyring
+      prettytable
+      lxml
+      matplotlib
+      colorspacious
+      jinja2
+      aiohttp
+      aiohttp-jinja2
+      python-bidi
+      arabic-reshaper
+      pillow
+      pydantic
+      babel
+      bleak
+      pyqt6
+      pyqt6-webengine
+      distro
+      secretstorage
+    ]
+  );
 in
-appimageTools.wrapType2 {
-  inherit pname version src;
+stdenv.mkDerivation {
+  inherit version pname;
+
+  src = fetchFromGitHub {
+    owner = "artisan-roaster-scope";
+    repo = "artisan";
+    rev = "v${version}";
+    sha256 = "sha256-EHq+tEXw7OQXFvHl4RPNpVz2xiYRiXDPJK0PHADnSzU=";
```

```suggestion
    sha256 = "sha256-IHaUeAlNVOrVbb376zvicm+Twwh1Yx54BFTEMt0MHm4=";
```

It looks like it has another hash now.